controller/src/main/java/org/jboss/as/controller/persistence/ConfigurationFile.java

```diff
@@ -68,21 +68,40 @@
     private final File configurationDir;
     private final String rawFileName;
     private final String bootFileName;
+    // File from which boot operations should be parsed; null if currently undetermined
     private volatile File bootFile;
+    /* Whether the next determination of the bootFile should use the .last file in history
+       instead of the {@link #mainFile}. Only relevant if {@link #persistOriginal} is false */
     private volatile boolean reloadUsingLast;
+    // Whether {@link #bootFile has been reset from its first value
+    private volatile boolean bootFileReset;
```

Comment:
Probably not any huge performance hit, but it likely doesn't need to be volatile since all access is guarded by `this`.